plugins/kubernetes/test/models/kubernetes/template_filler_test.rb

```diff
@@ -1069,6 +1069,39 @@ def secret_annotations(hash)
         end
       end
     end
+
+    describe 'istio sidecar injection' do
+      with_env ISTIO_INJECTION_SUPPORTED: "true"
+
+      before { doc.deploy_group_role.inject_istio_annotation = true }
+
+      let(:istio_annotation) { :'sidecar.istio.io/inject' }
+
+      let(:pod_annotation) { template.to_hash.dig(:spec, :template, :metadata, :annotations, istio_annotation) }
+      let(:pod_label) { template.to_hash.dig(:spec, :template, :metadata, :labels, istio_annotation) }
+      let(:resource_label) { template.to_hash.dig(:metadata, :labels, istio_annotation) }
+
+      it "modifies the Deployment" do
```

Comment:
What about StatefulSet, DaemonSet, Job, CronJob?

Reply:
The logic should work with any of them, since it uses the logic in TemplateFiller that applies to any k8s resource that have a pod template in it:

https://github.com/zendesk/samson/blob/ec06cc05bd97453fddb8acf7c8a02ca8a2345acc/plugins/kubernetes/app/models/kubernetes/template_filler.rb#L38-L57

It's a fair point that it would be a good idea to add tests to a few other resources though.

@grosser, are there any other tests for similar logic that iterate over all the types of resources that match `RoleConfigFile.primary?`

Reply:
use `pod_template` ... I'd just test with a single type or 2 (unless you add custom logic) ... the supported types are tested in `.primary?` tests 🤷‍♂